pkg/apis/kubeone/types.go

```diff
@@ -366,6 +368,30 @@ type SystemPackages struct {
 	ConfigureRepositories bool `json:"configureRepositories,omitempty"`
 }
 
+// ImageConfiguration controls which images are used for the core components
+type ImageConfiguration struct {
+	// Kubeadm image configuration
+	// Only ImageRepository is respected, ImageTag is ignored
+	Kubeadm ImageMeta `json:"kubeadm,omitempty"`
+	// Pause image configuration
+	// Respected only is special cases
+	Pause ImageMeta `json:"pause,omitempty"`
+	// CoreDNS image configuration
+	CoreDNS ImageMeta `json:"coreDNS,omitempty"`
+	// Etcd image configuration
+	Etcd ImageMeta `json:"etcd,omitempty"`
+	// metrics-server image configuration
+	MetricsServer ImageMeta `json:"metricsServer,omitempty"`
+}
+
+// ImageMeta is used to customize the image registry and the image tag
+type ImageMeta struct {
+	// ImageRepository customizes the registry/repository part of the image
+	ImageRepository string `json:"imageRepository,omitempty"`
```

Comment:
@kron4eg Should we call this `ImageRepository` or `ImageRegistry`? In the kubeadm API, it's called `ImageRepository`, but in the RegistryConfiguration API, we have `OverwriteRegistry`.